prisma/schema.prisma

```diff
@@ -0,0 +1,126 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+// ================ MODELS ================
+model User {
+  uid String @id
+
+  updatedAt DateTime @updatedAt @map("updated_at")
+
+  applications Application[]
+
+  @@map("users")
+}
+
+model Company {
+  id         Int     @id @default(autoincrement())
+  name       String
+  companyUrl String  @map("company_url")
+  isVerified Boolean @default(false) @map("is_verified")
+
+  updatedAt DateTime @updatedAt @map("updated_at")
+
+  roles Role[]
+
+  @@map("companies")
+}
+
+enum RoleType {
+  WINTER_INTERNSHIP
+  SPRING_INTERNSHIP
+  SUMMER_INTERNSHIP
+  FALL_INTERNSHIP
+  FULL_TIME
+}
+
+model Role {
+  id         Int      @id @default(autoincrement())
+  company    Company  @relation(fields: [companyId], references: [id])
+  companyId  Int      @map("company_id")
+  title      String
+  type       RoleType
+  year       Int
```

Comment:
Should we add a CHECK constraint here to check if this is really a valid year (current or future)? Not sure how it is done in Prisma.

Reply:
Do we really want this constraint though? I see potential use cases for allowing people to add applications for previous cycles.

Reply:
Okay I think we can leave it as it is for now.